amqp_log_writer.go

```diff
@@ -90,11 +92,10 @@ func (w *amqpLogWriter) Write(p []byte) (int, error) {
 
 	w.bytesWritten += len(p)
 	if w.bytesWritten > w.maxLength {
-		_, err := w.WriteAndClose([]byte(fmt.Sprintf("\n\nThe log length has exceeded the limit of %d MB (this usually means that the test suite is raising the same exception over and over).\n\nThe job has been terminated\n", w.maxLength/1000/1000)))
-		if err != nil {
-			logger.WithField("err", err).Error("couldn't write 'log length exceeded' error message to log")
-		}
-		return 0, ErrWrotePastMaxLogLength
+		logger.Info("wrote past maximum log length - cancelling context")
+		w.maxLengthReached = true
+		w.cancel()
```

Comment:
Could there be cases where `w.cancel` is not set?

Reply:
I think that if we care about termination for log length reasons, then it should be set. 
And since we can't properly process errors in this case, then crashing because it couldn't find the cancel func is the right thing to do - it will tell us that we've got something wrong. :)
Hopefully we'll catch these sorts of things in staging, because, despite what i said above, crashing in production is bad. 😅

Reply:
While this would probably not happen in practice, we do write to the log before setting cancel, e.g. when worker writes the booting messages. If for some reason we write too much data there, we could run into a crash here.

Not likely in practice but imo it would be good to check for nil just in case.